docs/development/kubernetes-clients.md

```diff
@@ -92,6 +92,12 @@ _Important characteristics of controller-runtime clients:_
   - The benefit is, that you don't lose updates to the API object and always have the last-known state in memory. Therefore, you don't have to read it again, e.g., for getting the current `resourceVersion` when working with [optimistic locking](#conflicts-concurrency-control-and-optimistic-locking), and thus minimize the chances for running into conflicts.
   - However, controllers *must not* use the same in-memory object concurrently in multiple goroutines. For example, decoding results from the API server in multiple goroutines into the same maps (e.g., labels, annotations) will cause panics because of "concurrent map writes". Also, reading from an in-memory API object in one goroutine while decoding into it in another goroutine will yield non-atomic reads, meaning data might be corrupt and represent a non-valid/non-existing API object.
   - Therefore, if you need to use the same in-memory object in multiple goroutines concurrently (e.g., shared state), remember to leverage proper synchronization techniques like channels, mutexes, `atomic.Value` and/or copy the object prior to use. The average controller however, will not need to share in-memory API objects between goroutines, and it's typically an indicator that the controller's design should be improved.
+- The client decoder erases the object's `TypeMeta` (`apiVersion` and `kind` fields) after retrieval from the API server, see [kubernetes/kubernetes#80609](https://github.com/kubernetes/kubernetes/issues/80609), [kubernetes-sigs/controller-runtime#1517](https://github.com/kubernetes-sigs/controller-runtime/issues/1517).
+  Unstructured and metadata-only requests objects are an exception to this, because the contained `TypeMeta` is the only way to identify the object's type.
```

Comment:
```suggestion
  Unstructured and metadata-only requests objects are an exception to this because the contained `TypeMeta` is the only way to identify the object's type.
```